internal/rbd/rbd_util.go

```diff
@@ -289,6 +300,50 @@ func (rv *rbdVolume) open() (*librbd.Image, error) {
 	return image, nil
 }
 
+// allocate uses the stripe-period of the image to fully allocate (thick
+// provision) the image.
+func (rv *rbdVolume) allocate(ctx context.Context) error {
+	util.DebugLog(ctx, "going to allocate %q with %d bytes, this may take time", rv.String(), rv.VolSize)
+
+	// We do not want to call discard, we really want to write zeros to get
+	// the allocation. This sets the option for the re-used connection, and
+	// all subsequent images that are opened. That is not a problem, as
+	// this is the only place images get written.
+	err := rv.conn.DisableDiscardOnZeroedWriteSame()
+	if err != nil {
+		return err
+	}
+
+	image, err := rv.open()
+	if err != nil {
+		return err
+	}
+	defer image.Close()
+
+	st, err := image.Stat()
+	if err != nil {
+		return err
+	}
+
+	sc, err := image.GetStripeCount()
+	if err != nil {
+		return err
+	}
+
+	// zeroBlock is the stripe-period: size of the object-size multiplied
+	// by the stripe-count
+	zeroBlock := make([]byte, sc*(1<<st.Order))
+
+	// the actual size of the image as available in the pool, can be
+	// marginally different from the requested image size
+	size := st.Size
+
+	s, err := image.WriteSame(0, size, zeroBlock, rados.OpFlagNone)
```

Comment:
just use st.Size without creating a new variable?

Reply:
sure, I can clean that up